lib/class-wp-rest-url-details-controller.php

```diff
@@ -137,6 +143,39 @@ public function parse_url_details( $request ) {
 		return apply_filters( 'rest_prepare_url_details', $response, $url, $request, $remote_url_response );
 	}
 
+	/**
+	 * Builds a new DOMXpath object from the given remote url response HTML
+	 * using DOMDocument.
+	 *
+	 * @param string $remote_url_response HTTP response body from the remote URL.
+	 * @return DOMXpath a new DOMXpath object from the remote URL response body.
+	 */
+	private function build_xpath_from_response( $remote_url_response ) {
+		// Initial DOMDocument.
+		$doc = new DOMDocument();
+
+		// phpcs:disable WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+		$doc->strictErrorChecking = false;
+		// phpcs:enable WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+
+		// Suppress warnings generated by loadHTML.
+		// see https://www.php.net/manual/en/domdocument.loadhtml.php#95463.
+		$errors = libxml_use_internal_errors( true );
+
+		// Ensure UTF-8 is respected by using 'mb_convert_encoding'.
+		$loaded_status = $doc->loadHTML( mb_convert_encoding( $remote_url_response, 'HTML-ENTITIES', 'UTF-8' ) );
+
+		libxml_use_internal_errors( $errors );
+
+		if ( false === $loaded_status ) {
+			return new WP_Error( 'rest_invalid_content', __( 'Unable to parse HTML response from remote URL.', 'gutenberg' ) );
+		}
+
+		$xpath = new DOMXpath( $doc );
+
+		return $xpath;
```

Comment:
Would it be easier to justreturn the `$doc` here? So devs are not limited to using XPath.